niceday-broker/index.js

```diff
@@ -29,16 +30,31 @@ function requestRasa(text, userId, callback) {
   xhr.send(data);
 }
 
+/**
+ * Send a message to a niceday recipient
+ * */
+function sendMessage(text, recipientId) {
+  chatSdk.sendTextMessage(recipientId, text).then((response) => {
+    console.log('Successfully sent the message', response);
+  });
+}
+
+function sleep(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
 /**
  * Handle the response from rasa, send each message to the Niceday user.
+ * We insert a delay in between messages, so the user has some time to read each message.
  * */
 function onRasaResponse() {
   if (this.readyState === 4 && this.status === 200) {
     const responseJson = JSON.parse(this.responseText);
-    responseJson.forEach((message) => {
-      chatSdk.sendTextMessage(parseInt(message.recipient_id, 10), message.text).then((response) => {
-        console.log('Successfully sent the message', response);
-      });
+    responseJson.forEach(async (message, i) => {
+      if (ENVIRONMENT === 'prod') {
+        await sleep(i * MESSAGE_DELAY);
+      }
```

Comment:
Yeah, good point. It will luckily not be a more than 5/6 messages. If we find out this this is not reliable we will improve it. 

_ And in this case I have no clear idea of what would happen if a new rasa response is produced before the end of the execution._ This would be kind of strange right? Because we send a request and only ever get one response back. What could happen is that a user sends another message, I have good hopes that will work, but maybe we need more asynchronous setup if it doesn't.

We should just do some stress testing before we start the clinical trial.